src/library/scala/collection/BitSet.scala

```diff
@@ -108,15 +108,22 @@ trait BitSetOps[+C <: BitSet with BitSetOps[C]]
   def iterator: Iterator[Int] = iteratorFrom(0)
 
   def iteratorFrom(start: Int): Iterator[Int] = new AbstractIterator[Int] {
-    private[this] var current = start
-    private[this] val end = nwords * WordLength
+    private[this] var currentPos = math.max(start >> LogWL, 0)
+    private[this] var currentWord = if (start > 0) word(currentPos) & (-1L << (start & (WordLength - 1))) else word(0)
     def hasNext: Boolean = {
-      while (current != end && !self.contains(current)) current += 1
-      current != end
+      while (currentWord == 0 && currentPos + 1 < nwords) {
+        currentPos += 1
+        currentWord = word(currentPos)
+      }
+      currentWord != 0
+    }
+    def next(): Int = {
+      if (hasNext) {
+        val bitPos = java.lang.Long.numberOfTrailingZeros(currentWord)
+        currentWord &= currentWord - 1
+        (currentPos << LogWL) + bitPos
+      } else Iterator.empty.next()
     }
```

Comment:
Cool stuff @linasm!

Does a minor tweaked solution like the following offer any benefits performance-wise?

WARNING: FOLLOWING CODE HAS NEVER BEEN COMPILED

```scala
def iteratorFrom(start: Int): Iterator[Int] = new AbstractIterator[Int] {
    private[this] final var currentPos = math.max(start >> LogWL, 0)
    private[this] final var currentWord = if (start > 0) word(currentPos) & (-1L << (start & (WordLength - 1))) else word(0)
    @tailrec private[this] final def nextWord(): Int = {
      if (currentWord == 0) {
        if (currentPos + 1 < nwords) {
          currentPos += 1
          currentWord = word(currentPos)
          nextWord()
        } else 0
      } else currentWord
    }
    final override def hasNext: Boolean = nextWord() != 0
    final override def next(): Int = {
      val word = nextWord()
      if (word != 0) {
        currentWord &= word - 1
        (currentPos << LogWL) + java.lang.Long.numberOfTrailingZeros(word)
      }
    }
  }
```

Reply:
@viktorklang thanks for your suggestions! I've tried benchmarking the suggested implementation, and it really gives a nice further improvement:
```
[info] Benchmark                           (spacing)  Mode  Cnt     Score    Error  Units
[info] BitSetIteratorBenchmark.iterateAll          0  avgt    6  3591.702 ± 30.049  ns/op
[info] BitSetIteratorBenchmark.iterateAll          1  avgt    6  1782.208 ± 19.453  ns/op
[info] BitSetIteratorBenchmark.iterateAll          3  avgt    6   896.987 ± 11.696  ns/op
[info] BitSetIteratorBenchmark.iterateAll         15  avgt    6   240.771 ±  2.595  ns/op
[info] BitSetIteratorBenchmark.iterateAll         63  avgt    6    77.982 ±  0.642  ns/op
[info] BitSetIteratorBenchmark.iterateAll        255  avgt    6    36.464 ±  0.428  ns/op
```
However, you helped me realize what can be improved in my implementation, and I was able to get basically the same (within ±1% margin) improvements, with less (and arguably simpler) code (updated the PR):
```
[info] Benchmark                           (spacing)  Mode  Cnt     Score    Error  Units
[info] BitSetIteratorBenchmark.iterateAll          0  avgt    6  3560.275 ± 65.491  ns/op
[info] BitSetIteratorBenchmark.iterateAll          1  avgt    6  1792.763 ± 13.405  ns/op
[info] BitSetIteratorBenchmark.iterateAll          3  avgt    6   908.467 ±  7.259  ns/op
[info] BitSetIteratorBenchmark.iterateAll         15  avgt    6   242.321 ±  3.470  ns/op
[info] BitSetIteratorBenchmark.iterateAll         63  avgt    6    78.177 ±  0.852  ns/op
[info] BitSetIteratorBenchmark.iterateAll        255  avgt    6    36.429 ±  0.200  ns/op
```
Please let me know what you think, thanks.